### ISSUE TITLE
vcd_edgegateway: Entity not found when more than 25 edge gatways exist

### ISSUE THREAD

**Issue by CmdrSharp** [vmware/terraform-provider-vcd#658]
### Terraform Version
Terraform v0.13.5

### Affected Resource(s)
- vcd_edgegateway

Unsure whether this could affect other resources as well.

### Terraform Configuration Files
```json
{
    "resource": {
        "vcd_network_routed": {
            "NetworkNameCensored": {
                "org": "OrgNameCensored",
                "vdc": "VdcNameCensored",
                "name": "NetworkNameCensored",
                "gateway": "IPCensored",
                "netmask": "255.255.255.240",
                "shared": true,
                "dns1": null,
                "dns2": null,
                "dns_suffix": null,
                "edge_gateway": "EdgeNameCensored",
                "static_ip_pool": {
                    "start_address": "IPCensored",
                    "end_address": "IPCensored"
                }
            }
        }
    }
}
```

### Debug Output
Leaving the only relevant output here
```
Error: [routed network read] error retrieving edge gateway PBX-SEFRE140-TESG-Arvika-OAM: [ENF] entity not found
```

### Panic Output
Panic only occurs on creation, not on import. On creating, it would produce the following
```
vcd_network_routed.NameCensored: Creating...
Error: rpc error: code = Unavailable desc = transport is closing

panic: edge gateway 'EdgeCensored' not found: unable to find edge gateway: [ENF] entity not found
```

### Expected Behavior
Network should have been created/imported as the edge gateway does indeed exist.

### Actual Behavior
Error or panic resulting in the resource not being created/imported.

### Steps to Reproduce
1. Attempt to work toward a dependant resource that is outside of the 25 list limit of fetched edge gateways.

### Important Factoids
I believe this is a bug in [go-vcloud-director](https://github.com/vmware/go-vcloud-director/issues/363) and have as such submitted an issue there as well. 
I just figure that this should probably be tracked here as well.

### References
- [go-vcloud-director #363](https://github.com/vmware/go-vcloud-director/issues/363)

**Comment by Didainius** [comment]
@CmdrSharp ,
Thanks for this issue - not a fix, but - if you don't use DHCP I'd suggest to use `vcd_network_routed_v2` which shouldn't have such issue.

**Comment by CmdrSharp** [comment]
@Didainius Thanks for the suggestion! How would other existing resources handle a switch to this? Would they require replacing or are the resources compatible somehow?

**Comment by dataclouder** [comment]
Fixed in PR #660
